src/main/java/uk/gov/hmcts/darts/common/service/impl/AutomatedTaskServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+        if (automatedTaskEntity.isPresent()) {
+            cronExpression = automatedTaskEntity.get().getCronExpression();
+            log.debug("{} cron expression: {}", taskName, cronExpression);
+        }
+        return cronExpression;
+    }
+
+    @Override
+    public void updateAutomatedTaskCronExpression(String taskName, String cronExpression) {
+        Optional<AutomatedTaskEntity> automatedTaskEntity = getAutomatedTaskEntityByTaskName(taskName);
+        if (automatedTaskEntity.isPresent()) {
+            AutomatedTaskEntity automatedTask = automatedTaskEntity.get();
+            if (CronExpression.isValidExpression(cronExpression)) {
+                automatedTask.setCronExpression(cronExpression);
+                automatedTaskRepository.saveAndFlush(automatedTask);
+                log.debug("Updated the task {} with cron expression {}", taskName, cronExpression);
+            }  else {
+                log.error("Unable to update the task {} with cron expression {}", taskName, cronExpression);
+                throw new DartsApiException(INVALID_CRON_EXPRESSION);
+            }
+        }  else {
+            log.error("Unable to update the task {} with cron expression {}", taskName, cronExpression);
+            throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+    }
+
+    @Override
+    public void loadTaskByName(String taskName) {
+        switch (taskName) {
+            case ProcessDailyListAutomatedTask.PROCESS_DAILY_LIST_TASK_NAME -> loadProcessDailyListAutomatedTask();
+            default -> throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+    }
+
+    public void loadProcessDailyListAutomatedTask() {
+        ProcessDailyListAutomatedTask processDailyListAutomatedTask = new ProcessDailyListAutomatedTask();
+        Trigger trigger = createAutomatedTaskTrigger(processDailyListAutomatedTask);
+        taskScheduler.schedule(processDailyListAutomatedTask, trigger);
+    }
+
+    private void loadProcessDailyListAutomatedTask(ScheduledTaskRegistrar taskRegistrar) {
+        ProcessDailyListAutomatedTask processDailyListAutomatedTask = new ProcessDailyListAutomatedTask();
+        Trigger trigger = createAutomatedTaskTrigger(processDailyListAutomatedTask);
+        taskRegistrar.addTriggerTask(processDailyListAutomatedTask, trigger);
+    }
+
+    @Override
+    public boolean cancelAutomatedTask(String taskName, boolean mayInterruptIfRunning) {
+        boolean result = false;
+        Set<ScheduledTask> scheduledTasks = taskHolder.getScheduledTasks();
+        for (ScheduledTask scheduledTask : scheduledTasks) {
+            Task task = scheduledTask.getTask();
+            if (task instanceof TriggerTask) {
+                TriggerTask triggerTask = (TriggerTask) task;
+                if (cancelTriggerTask(taskName, scheduledTask, triggerTask, mayInterruptIfRunning)) {
+                    result = true;
+                    break;
+                }
+            }
```

Comment:
To improve readability, suggest eliminating the boolean flags by simply returning early, e.g.:
```java
    @Override
    public boolean cancelAutomatedTask(String taskName, boolean mayInterruptIfRunning) {
        Set<ScheduledTask> scheduledTasks = taskHolder.getScheduledTasks();
        for (ScheduledTask scheduledTask : scheduledTasks) {
            Task task = scheduledTask.getTask();
            if (task instanceof TriggerTask triggerTask) {
                if (cancelTriggerTask(taskName, scheduledTask, triggerTask, mayInterruptIfRunning)) {
                    return true;
                }
            }
        }
        return false;
    }
```